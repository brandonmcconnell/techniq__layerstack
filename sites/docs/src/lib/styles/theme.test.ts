import { describe, it, expect } from 'vitest';
import { flattenThemeColors, processThemeColors } from './theme.js';
import tailwindColors from 'tailwindcss/colors.js';

describe('flattenThemeColors', () => {
  it('Flatten recursive colors object', () => {
    const actual = flattenThemeColors({
      // Simple key-value pair
      primary: '#ff0000',
      dark: {
        // Default value
        DEFAULT: '#000',
        // Nested once
        blue: '#0000ff',
        red: {
          // Nested twice
          500: '#f00',
          // Multiple nested properties
          content: 'white',
        },
      },
      light: {
        // Should not be used, since objects do not guarantee respecting insertion, but it should work without any error
        green: { 100: '#0f0' },
        'green-100': '#0f0',
      },
    });
    const expected = {
      primary: '#ff0000',
      dark: '#000',
      'dark-blue': '#0000ff',
      'dark-red-500': '#f00',
      'dark-red-content': 'white',
      'light-green-100': '#0f0',
    };

    expect(actual).toEqual(expected);
  });
});

describe('processThemeColors', () => {
  it('Process theme colors recursive colors object', () => {
    const actual = processThemeColors(
      {
        // overriding semantic colors
        primary: {
          ...tailwindColors.blue,
          content: 'white',
          DEFAULT: tailwindColors.blue[600],
        },
        secondary: { ...tailwindColors.orange, content: 'white' },
        accent: { ...tailwindColors.fuchsia, content: 'white' },
        neutral: { ...tailwindColors.gray, content: 'white' },
        // overriding state colors
        info: tailwindColors.blue,
        success: tailwindColors.green,
        warning: tailwindColors.amber,
        danger: tailwindColors.red,
        // overriding surface colors
        surface: {
          100: 'white',
          200: tailwindColors.gray[200],
          300: tailwindColors.gray[300],
          content: tailwindColors.gray[900],
        },
      },
      'hsl'
    );

    expect(actual).toMatchSnapshot();
  });
});
