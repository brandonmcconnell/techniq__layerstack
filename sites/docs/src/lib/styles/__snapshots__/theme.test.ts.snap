// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`processThemeColors > Process theme colors recursive colors object 1`] = `
{
  "--color-accent": "292.1893 84.0796% 60.5882%",
  "--color-accent-100": "286.9565 100% 95.4902%",
  "--color-accent-200": "288.2609 95.8333% 90.5882%",
  "--color-accent-300": "291.1111 93.1034% 82.9412%",
  "--color-accent-400": "292.0313 91.4286% 72.549%",
  "--color-accent-50": "289.0909 100% 97.8431%",
  "--color-accent-500": "292.1893 84.0796% 60.5882%",
  "--color-accent-600": "293.4104 69.4779% 48.8235%",
  "--color-accent-700": "294.6939 72.4138% 39.8039%",
  "--color-accent-800": "295.4237 70.2381% 32.9412%",
  "--color-accent-900": "296.7033 63.6364% 28.0392%",
  "--color-accent-content": "0 0% 100%",
  "--color-danger": "0 84.2365% 60.1961%",
  "--color-danger-100": "0 93.3333% 94.1176%",
  "--color-danger-200": "0 96.2963% 89.4118%",
  "--color-danger-300": "0 93.5484% 81.7647%",
  "--color-danger-400": "0 90.604% 70.7843%",
  "--color-danger-50": "0 85.7143% 97.2549%",
  "--color-danger-500": "0 84.2365% 60.1961%",
  "--color-danger-600": "0 72.2222% 50.5882%",
  "--color-danger-700": "0 73.7089% 41.7647%",
  "--color-danger-800": "0 70% 35.2941%",
  "--color-danger-900": "0 62.8205% 30.5882%",
  "--color-danger-content": "0 85.7211% 4.1843%",
  "--color-info": "217.2193 91.2195% 59.8039%",
  "--color-info-100": "214.2857 94.5946% 92.7451%",
  "--color-info-200": "213.3333 96.9231% 87.2549%",
  "--color-info-300": "211.6981 96.3636% 78.4314%",
  "--color-info-400": "213.1169 93.9024% 67.8431%",
  "--color-info-50": "213.75 100% 96.8627%",
  "--color-info-500": "217.2193 91.2195% 59.8039%",
  "--color-info-600": "221.2121 83.1933% 53.3333%",
  "--color-info-700": "224.2781 76.3265% 48.0392%",
  "--color-info-800": "225.931 70.7317% 40.1961%",
  "--color-info-900": "224.4444 64.2857% 32.9412%",
  "--color-info-content": "225.5179 89.4756% 4.2952%",
  "--color-neutral": "220 8.9362% 46.0784%",
  "--color-neutral-100": "220 14.2857% 95.8824%",
  "--color-neutral-200": "220 13.0435% 90.9804%",
  "--color-neutral-300": "216 12.1951% 83.9216%",
  "--color-neutral-400": "217.8947 10.6145% 64.902%",
  "--color-neutral-50": "210 20% 98.0392%",
  "--color-neutral-500": "220 8.9362% 46.0784%",
  "--color-neutral-600": "215 13.7931% 34.1176%",
  "--color-neutral-700": "216.9231 19.1176% 26.6667%",
  "--color-neutral-800": "215 27.907% 16.8627%",
  "--color-neutral-900": "220.9091 39.2857% 10.9804%",
  "--color-neutral-content": "0 0% 100%",
  "--color-primary": "221.2121 83.1933% 53.3333%",
  "--color-primary-100": "214.2857 94.5946% 92.7451%",
  "--color-primary-200": "213.3333 96.9231% 87.2549%",
  "--color-primary-300": "211.6981 96.3636% 78.4314%",
  "--color-primary-400": "213.1169 93.9024% 67.8431%",
  "--color-primary-50": "213.75 100% 96.8627%",
  "--color-primary-500": "217.2193 91.2195% 59.8039%",
  "--color-primary-600": "221.2121 83.1933% 53.3333%",
  "--color-primary-700": "224.2781 76.3265% 48.0392%",
  "--color-primary-800": "225.931 70.7317% 40.1961%",
  "--color-primary-900": "224.4444 64.2857% 32.9412%",
  "--color-primary-content": "0 0% 100%",
  "--color-secondary": "24.5815 94.9791% 53.1373%",
  "--color-secondary-100": "34.2857 100% 91.7647%",
  "--color-secondary-200": "32.1429 97.6744% 83.1373%",
  "--color-secondary-300": "30.6569 97.1631% 72.3529%",
  "--color-secondary-400": "27.0157 95.9799% 60.9804%",
  "--color-secondary-50": "33.3333 100% 96.4706%",
  "--color-secondary-500": "24.5815 94.9791% 53.1373%",
  "--color-secondary-600": "20.5405 90.2439% 48.2353%",
  "--color-secondary-700": "17.4725 88.3495% 40.3922%",
  "--color-secondary-800": "15 79.0698% 33.7255%",
  "--color-secondary-900": "15.283 74.6479% 27.8431%",
  "--color-secondary-content": "0 0% 100%",
  "--color-success": "142.0859 70.5628% 45.2941%",
  "--color-success-100": "140.625 84.2105% 92.549%",
  "--color-success-200": "141 78.9474% 85.098%",
  "--color-success-300": "141.7143 76.6423% 73.1373%",
  "--color-success-400": "141.8919 69.1589% 58.0392%",
  "--color-success-50": "138.4615 76.4706% 96.6667%",
  "--color-success-500": "142.0859 70.5628% 45.2941%",
  "--color-success-600": "142.1277 76.2162% 36.2745%",
  "--color-success-700": "142.4299 71.8121% 29.2157%",
  "--color-success-800": "142.7848 64.2276% 24.1176%",
  "--color-success-900": "143.8095 61.165% 20.1961%",
  "--color-success-content": "131.0117 94.2327% 2.8668%",
  "--color-surface-100": "0 0% 100%",
  "--color-surface-200": "220 13.0435% 90.9804%",
  "--color-surface-300": "216 12.1951% 83.9216%",
  "--color-surface-content": "220.9091 39.2857% 10.9804%",
  "--color-warning": "37.6923 92.126% 50.1961%",
  "--color-warning-100": "48 96.4912% 88.8235%",
  "--color-warning-200": "48 96.6387% 76.6667%",
  "--color-warning-300": "45.9429 96.6851% 64.5098%",
  "--color-warning-400": "43.2558 96.4126% 56.2745%",
  "--color-warning-50": "48 100% 96.0784%",
  "--color-warning-500": "37.6923 92.126% 50.1961%",
  "--color-warning-600": "32.1327 94.6188% 43.7255%",
  "--color-warning-700": "25.9649 90.4762% 37.0588%",
  "--color-warning-800": "22.7273 82.5% 31.3725%",
  "--color-warning-900": "21.7143 77.7778% 26.4706%",
  "--color-warning-content": "26.0785 99.1481% 4.0603%",
  "accent-950": "#4a044e",
  "color-scheme": "light",
  "danger-950": "#450a0a",
  "info-950": "#172554",
  "neutral-950": "#030712",
  "primary-950": "#172554",
  "secondary-950": "#431407",
  "success-950": "#052e16",
  "warning-950": "#451a03",
}
`;
